### PR TITLE
grab all the page data they give us

### PR DESCRIPTION
**Pull request by ktbartholomew**
I'm ashamed it took me this long to change 5 lines of code.

@smashwilson I feel like this is pretty sane and won't break anything we have existing. Anything look out of place to you?

### CODE REVIEW

File: lib/preparermd/plugins/metadata_envelopes.rb [hunk 1]
```diff
@@ -38,12 +38,11 @@ def render_json(page, site)
 
       output = page.to_liquid
 
-      envelope = {
-        title: output["title"],
-        body: output["content"],
-        layout_key: layout,
-        categories: output["categories"] || []
-      }
+      envelope = page.data;
```
Comment (smashwilson) [inline lib/preparermd/plugins/metadata_envelopes.rb]: 1. No semicolons in Ruby :wink:
2. I'm wondering if this should be `page.data.dup` so we don't modify the original data structure in unexpected ways? If you were able to run this locally without issues, it's probably fine.

**Comment by smashwilson** [comment]
> I'm ashamed it took me this long to change 5 lines of code.

Finding the _right_ 5 lines can take a lot longer!

I'm generally fine with this, as long as we maintain a consistent [core schema](http://deconst.horse/developing/envelope/) for the envelopes that the content service accepts. (Validating that on content post is an idea I've had on the backburner for a while now.)

The danger is that, by using these attributes, we'll be coupling the template used to render a piece of content more tightly with the preparer most recently used to submit it. If we ever migrate content from Jekyll to Sphinx or Sphinx to something else, the variables visible to the template will change, and template rendering can fail. You'd also need to be careful which attributes you make use of if you want a template to be portable across content from different sources, especially within common fragments included in different places.

How would you feel about namespacing these beneath an intermediate key, like `jekyll`? That lets you quickly find them with a Ctrl-f or a grep. You could even only use them in `{{ if envelope.jekyll }}` guard blocks to let the template degrade gracefully if we mix things up. We can always promote common, useful ones into the top-level schema (and therefore ensure that they're always named consistently and so forth).

**Comment by ktbartholomew** [comment]
In my (conceptual) templates, I'm already working from the assumption that metadata like this won't always be available. They're all strictly optional.

I agree that moving them into a namespace would tidy things up a bit (but I shot down my own idea for over-architecting). I think naming the namespace key after the preparer would cause some confusion on output, where ideally we don't have to think about which preparer submitted the content we're trying to render.

What about a namespace key called simply `meta`?

**Comment by smashwilson** [comment]
> In my (conceptual) templates, I'm already working from the assumption that metadata like this won't always be available. They're all strictly optional.

Sure - I just wanted it to be obvious at a glance which properties were guaranteed to be present, and which were preparer-dependent.

> I agree that moving them into a namespace would tidy things up a bit (but I shot down my own idea for over-architecting). I think naming the namespace key after the preparer would cause some confusion on output, where ideally we don't have to think about which preparer submitted the content we're trying to render.
> 
> What about a namespace key called simply meta?

To me, using optional attributes like this _does_ bind you to the preparer. Jekyll will produce a consistent, but different, set of attributes than Sphinx.

Hmm, dropping them under `meta` or `extra` or something like that would be okay, though. It would still separate the mandatory from the optional ones. Yeah, let's go with that :grin:

**Pull request merged**